From 6bf95871549e858213aeba2f7f33aab1467e6b80 Mon Sep 17 00:00:00 2001
From: razorloves <razorloves@gmail.com>
Date: Wed, 30 Jan 2019 01:28:48 -0600
Subject: [PATCH 1/2] SUW: Update wizard scripts for Pie

The google suw flow has changed a little bit and added some new flow
choices in Pie, so our scripts need to be updated to match it. This
fixes the restore choice screen being shown twice.

Also reorganized the order of the wizard action results to move the
resultcode from the beginning to the end. This is a noop change that's
just to bring it inline with google's suw scripts.

Change-Id: Ia3ac9d1fdae62132c377e8ac2f767a795790d092
---
 res/raw/wizard_script.xml      | 25 ++++++++++++++++---------
 res/raw/wizard_script_user.xml | 22 ++++++++++++++--------
 2 files changed, 30 insertions(+), 17 deletions(-)

diff --git a/res/raw/wizard_script.xml b/res/raw/wizard_script.xml
index fe13af5..cebdf0f 100644
--- a/res/raw/wizard_script.xml
+++ b/res/raw/wizard_script.xml
@@ -2,7 +2,7 @@
 
 <!--
     Copyright (c) 2014 Google Inc.
-    Copyright (C) 2017 The LineageOS Project
+    Copyright (C) 2017,2019 The LineageOS Project
 
     Licensed under the Apache License, Version 2.0 (the "License");
     you may not use this file except in compliance with the License.
@@ -29,8 +29,8 @@
   xmlns:wizard="http://schemas.android.com/apk/res/com.google.android.setupwizard">
     <WizardAction wizard:uri="intent:#Intent;action=com.android.setupwizard.OEM_PRE_SETUP;end" id="oem_pre_setup" />
     <WizardAction wizard:uri="intent:#Intent;action=com.android.setupwizard.WELCOME;end" id="welcome">
-        <result wizard:resultCode="101" wizard:action="check_user_unlock_qr" wizard:name="start_qr_provision" />
-        <result wizard:resultCode="111" wizard:action="check_user_unlock_dpm_user_complete" wizard:name="dpm_user_complete" />
+        <result wizard:action="check_user_unlock_qr" wizard:name="start_qr_provision" wizard:resultCode="101" />
+        <result wizard:action="check_user_unlock_dpm_user_complete" wizard:name="dpm_user_complete" wizard:resultCode="111" />
         <result wizard:action="check_user_unlock" />
     </WizardAction>
     <WizardAction wizard:uri="intent:#Intent;action=com.android.setupwizard.CHECK_USER_UNLOCK;end" id="check_user_unlock_qr">
@@ -41,36 +41,43 @@
     </WizardAction>
     <WizardAction wizard:uri="intent:#Intent;action=com.android.setupwizard.CHECK_USER_UNLOCK;end" id="check_user_unlock" />
     <WizardAction wizard:uri="intent:#Intent;action=com.android.setupwizard.SIM_MISSING;end" id="sim_missing">
-        <result wizard:resultCode="101" wizard:action="esim_intro" wizard:name="esim" />
+        <result wizard:action="esim_intro" wizard:name="esim" wizard:resultCode="101" />
         <result wizard:action="carrier_setup" />
     </WizardAction>
     <WizardAction wizard:uri="intent:#Intent;action=com.android.setupwizard.ESIM_INTRO;end" id="esim_intro" />
     <WizardAction wizard:uri="intent:#Intent;action=com.android.setupwizard.CARRIER_SETUP;end" id="carrier_setup" />
     <WizardAction wizard:uri="intent:#Intent;action=com.android.setupwizard.SIM_SETUP;end" id="sim_setup" />
     <WizardAction wizard:uri="intent:#Intent;action=com.google.android.setupwizard.DEVICE_OWNER_WARNING;end" id="device_owner_warning">
-        <result wizard:resultCode="1" wizard:action="check_frp" wizard:name="skip" />
+        <result wizard:action="check_frp" wizard:name="skip" wizard:resultCode="1" />
     </WizardAction>
     <WizardAction wizard:uri="intent:#Intent;action=com.google.android.setupwizard.FACTORY_RESET;end" id="factory_reset" />
     <WizardAction wizard:uri="intent:#Intent;action=com.google.android.setupwizard.CHECK_FRP;end" id="check_frp" />
     <WizardAction wizard:script="android.resource://com.google.android.setupwizard/xml/wizard_script_connect_and_update_flow" id="connect_and_update">
-        <result wizard:resultCode="1" wizard:action="no_network_flow" wizard:name="no_connection" />
+        <result wizard:action="no_network_flow" wizard:name="no_connection" wizard:resultCode="1" />
     </WizardAction>
     <WizardAction wizard:script="android.resource://com.google.android.setupwizard/xml/wizard_script_zero_touch_flow" id="zero_touch">
-        <result wizard:resultCode="111" wizard:action="oem_post_setup" wizard:name="dpm_user_complete" />
+        <result wizard:action="post_dpm_user_flow" wizard:name="dpm_user_complete" wizard:resultCode="111" />
     </WizardAction>
     <WizardAction wizard:uri="intent:#Intent;action=com.android.setupwizard.FLOW_CHOICE;end" id="flow_choice">
-        <result wizard:resultCode="1" wizard:action="setup_as_new_flow" wizard:name="skip" />
-        <result wizard:resultCode="101" wizard:action="exit" wizard:name="demo_mode_flow" />
+        <result wizard:action="setup_as_new_flow" wizard:name="skip" wizard:resultCode="1" />
+        <result wizard:action="exit" wizard:name="demo_mode_flow" wizard:resultCode="101" />
+        <result wizard:action="unified_restore_flow" wizard:name="unified_restore_flow" wizard:resultCode="102" />
     </WizardAction>
     <WizardAction wizard:script="android.resource://com.google.android.setupwizard/xml/wizard_script_restore_flow" id="restore_flow">
         <result wizard:action="oem_post_setup" />
     </WizardAction>
+    <WizardAction wizard:script="android.resource://com.google.android.setupwizard/xml/wizard_script_unified_restore_flow" id="unified_restore_flow">
+        <result wizard:action="oem_post_setup" />
+    </WizardAction>
     <WizardAction wizard:script="android.resource://com.google.android.setupwizard/xml/wizard_script_setup_as_new_flow" id="setup_as_new_flow">
         <result wizard:action="oem_post_setup" />
     </WizardAction>
     <WizardAction wizard:script="android.resource://com.google.android.setupwizard/xml/wizard_script_no_network_flow" id="no_network_flow">
         <result wizard:action="oem_post_setup" />
     </WizardAction>
+    <WizardAction wizard:script="android.resource://com.google.android.setupwizard/xml/wizard_script_post_dpm_user_flow" id="post_dpm_user_flow">
+        <result wizard:action="oem_post_setup" />
+    </WizardAction>
     <WizardAction wizard:script="android.resource://com.google.android.setupwizard/xml/wizard_script_qr_provision_flow" id="qr_provision_flow" />
     <WizardAction wizard:uri="intent:#Intent;action=org.lineageos.setupwizard.LINEAGE_SETTINGS;end" id="oem_post_setup">
         <result wizard:action="finish" />
diff --git a/res/raw/wizard_script_user.xml b/res/raw/wizard_script_user.xml
index 54dd967..a8a627f 100644
--- a/res/raw/wizard_script_user.xml
+++ b/res/raw/wizard_script_user.xml
@@ -2,6 +2,7 @@
 
 <!--
     Copyright (c) 2014 Google Inc.
+    Copyright (C) 2017,2019 The LineageOS Project
 
     Licensed under the Apache License, Version 2.0 (the "License");
     you may not use this file except in compliance with the License.
@@ -27,7 +28,7 @@
   xmlns:wizard="http://schemas.android.com/apk/res/com.google.android.setupwizard">
     <WizardAction wizard:uri="intent:#Intent;action=com.android.setupwizard.OEM_PRE_SETUP;end" id="oem_pre_setup" />
     <WizardAction wizard:uri="intent:#Intent;action=com.android.setupwizard.USER_WARNING;end" id="secondary_user_warning">
-        <result wizard:resultCode="111" wizard:action="check_user_unlock_dpm_user_complete" wizard:name="dpm_user_complete" />
+        <result wizard:action="check_user_unlock_dpm_user_complete" wizard:name="dpm_user_complete" wizard:resultCode="111" />
         <result wizard:action="check_user_unlock" />
     </WizardAction>
     <WizardAction wizard:uri="intent:#Intent;action=com.android.setupwizard.CHECK_USER_UNLOCK;end" id="check_user_unlock_dpm_user_complete">
@@ -35,25 +36,30 @@
     </WizardAction>
     <WizardAction wizard:uri="intent:#Intent;action=com.android.setupwizard.CHECK_USER_UNLOCK;end" id="check_user_unlock" />
     <WizardAction wizard:uri="intent:#Intent;action=com.android.setupwizard.NETWORK_SETTINGS;end" id="network_settings">
-        <result wizard:resultCode="102" wizard:action="wifi_settings" wizard:name="see_all_wifi" />
-        <result wizard:resultCode="1" wizard:action="no_account_flow" wizard:name="skip" />
+        <result wizard:action="wifi_settings" wizard:name="see_all_wifi" wizard:resultCode="102" />
+        <result wizard:action="network_check" wizard:name="skip" wizard:resultCode="1" />
         <result wizard:action="captive_portal" />
     </WizardAction>
     <WizardAction wizard:uri="intent:#Intent;action=com.android.setupwizard.WIFI_SETTINGS;end" id="wifi_settings">
-        <result wizard:resultCode="1" wizard:action="no_account_flow" wizard:name="skip" />
+        <result wizard:action="network_check" wizard:name="skip" wizard:resultCode="1" />
     </WizardAction>
     <WizardAction wizard:uri="intent:#Intent;action=com.android.setupwizard.CAPTIVE_PORTAL;end" id="captive_portal" />
     <WizardAction wizard:uri="intent:#Intent;action=com.google.android.setupwizard.GMS_CHECKIN;end" id="gms_checkin" />
+    <WizardAction wizard:uri="intent:#Intent;action=com.google.android.setupwizard.EARLY_UPDATE;end" id="early_update" />
+    <WizardAction wizard:uri="intent:#Intent;action=com.google.android.setupwizard.NETWORK_CHECK;end" id="network_check">
+        <result wizard:action="no_account_flow" wizard:name="skip" wizard:resultCode="1" />
+    </WizardAction>
+    <WizardAction wizard:script="android.resource://com.google.android.setupwizard/xml/wizard_script_rollback_auth_early_update_flow" id="rollback_auth_early_update" />
     <WizardAction wizard:uri="intent:#Intent;action=com.google.android.setupwizard.LOAD_ADD_ACCOUNT_INTENT;end" id="load_account_intent" />
     <WizardAction wizard:uri="intent:#Intent;action=com.google.android.setupwizard.ACCOUNT_SETUP;end" id="account_setup">
-        <result wizard:resultCode="1" wizard:action="no_account_flow" wizard:name="skip" />
-        <result wizard:resultCode="111" wizard:action="oem_post_setup" wizard:name="dpm_user_complete" />
+        <result wizard:action="no_account_flow" wizard:name="skip" wizard:resultCode="1" />
+        <result wizard:action="oem_post_setup" wizard:name="dpm_user_complete" wizard:resultCode="111" />
     </WizardAction>
     <WizardAction wizard:uri="intent:#Intent;action=com.google.android.setupwizard.GMS_ACCOUNT_CHECKIN;end" id="gms_account_checkin">
-        <result wizard:resultCode="1" wizard:action="mfm_check" wizard:name="skip" />
+        <result wizard:action="mfm_check" wizard:name="skip" wizard:resultCode="1" />
     </WizardAction>
     <WizardAction wizard:uri="intent:#Intent;action=com.google.android.setupwizard.ACCOUNT_CHECK;end" id="mfm_check">
-        <result wizard:resultCode="1" wizard:action="no_account_flow" wizard:name="skip" />
+        <result wizard:action="no_account_flow" wizard:name="skip" wizard:resultCode="1" />
     </WizardAction>
     <WizardAction wizard:script="android.resource://com.google.android.setupwizard/xml/wizard_script_user_account_flow" id="account_flow">
         <result wizard:action="oem_post_setup" />
-- 
2.17.1

